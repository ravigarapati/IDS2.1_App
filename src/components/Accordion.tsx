import React, {useState} from 'react';
import {StyleSheet, TouchableWithoutFeedback, View} from 'react-native';
import {Icons} from '../utils/icons';
import {BoschIcon, CustomText} from '.';
import {Colors} from '../styles';
type accordionProps = {
  title: string;
  disabled: boolean;
  children: any;
  index: number;
};
export default function Accordion({
  title,
  disabled,
  index,
  children,
}: accordionProps) {
  const [open, setOpen] = useState(false);
  const borderTop = index === 0 ? {borderTopWidth: 1} : {borderTopWidth: 0};
  return (
    <>
      <View>
        <View style={[styles.title, borderTop]}>
          <CustomText size={16} font="medium" text={title} />
          <TouchableWithoutFeedback
            hitSlop={{left: 20, right: 20, top: 20, bottom: 20}}
            onPress={() => (disabled ? {} : setOpen(!open))}
            accessibilityLabel={open ? (title + 'expanded') : (title + 'collapsed')}
            accessibilityState={{expanded: open}}>
            <BoschIcon
              color={disabled ? Colors.grayDisabled : Colors.darkBlue}
              name={open ? Icons.up : Icons.down}
              size={30}
              style={{height: 30}}
            />
          </TouchableWithoutFeedback>
        </View>
      </View>
      {open && <View style={styles.bodyBackground}>{children}</View>}
    </>
  );
}

const styles = StyleSheet.create({
  title: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderWidth: 1,
    borderColor: Colors.mediumGray,
    padding: 20,
    flex: 1,
  },
  bodyBackground: {
    backgroundColor: Colors.white,
    borderColor: Colors.mediumGray,
    padding: 20,
    borderBottomWidth: 1,
    borderLeftWidth: 1,
    borderRightWidth: 1,
  },
});
