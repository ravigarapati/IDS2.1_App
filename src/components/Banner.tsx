import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Dictionary } from '../utils/dictionary';
import BoschIcon from './BoschIcon';
import CustomText from './CustomText';
import InfoTooltip from './InfoTooltip';
import Link from './Link';

type dataProps = {
  data: {
    background: string;
    iconColor: string;
    iconName: string;
    text: string;
    tooltipText?: string;
    warningBanner?: boolean;
  };
  navigation?: any;
};
export default function Banner({ data, navigation }: dataProps) {
  return (
    <View
      style={[
        styles.container,
        {
          backgroundColor: data.background,
        },
      ]}>
      <View style={[
        styles.flexRow,
        data.warningBanner ? styles.warningBannerFlexRow : styles.iconFlexRow]}>
        <BoschIcon name={data.iconName} color={data.iconColor}
          style={data.warningBanner ? styles.warningBannerIcon : styles.icon} />
        <CustomText
          color={data.iconColor}
          style={[styles.textPadding]}
          text={data.text}
          align="left"
        />
      </View>
      {data.tooltipText && (
        <View>
          <InfoTooltip positionVertical="bottom" text={data.tooltipText} />
        </View>
      )}
      {data.text.includes(Dictionary.systemData.statusAlert) && (
        <Link
          text="Service"
          type="arrow"
          onPress={() => {
            navigation.navigate('Service');
          }}
        />
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 20,
  },
  textPadding: {
    paddingLeft: 10,
  },
  flexRow: {
    flex: 1,
    flexDirection: 'row',
  },
  warningBannerIcon: {
    marginTop: 5,
    fontSize: 32,
    height: 32,
  },
  warningBannerFlexRow: {
    alignItems: 'flex-start',
    paddingVertical: 20,
  },
  iconFlexRow: {
    alignItems: 'center',
    paddingVertical: 10,
  },
  icon: {
    fontSize: 26,
    height: 26,
  },
});
