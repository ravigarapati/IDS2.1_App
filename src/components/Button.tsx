import React from 'react';
import {View, TouchableHighlight, StyleSheet} from 'react-native';
import BoschIcon from './BoschIcon';
import {Colors} from '../styles';
import CustomText from './CustomText';
type Button = {
  text: string | number;
  type: string;
  disabled?: boolean;
  onPress?: () => void;
  style?: {};
  commonStyle?: {};
  textStyle?: {};
  icon?: string;
  accessibilityLabelText?: string;
  accessibilityHintText?: string;
  testID?: string;
};
export default function Button({
  text,
  onPress,
  type = 'primary',
  disabled = false,
  style,
  textStyle,
  commonStyle = {},
  icon = '',
  accessibilityLabelText = '',
  accessibilityHintText = '',
  testID = '',
}: Button) {
  const isDisabled = disabled;
  const disabledView =
    type === 'primary'
      ? {backgroundColor: Colors.grayDisabled}
      : {backgroundColor: Colors.white};

  var [isPress, setIsPress] = React.useState(false);

  var touchPropsPrimary = {
    activeOpacity: 1,
    underlayColor: Colors.blueOnPress,
    onPressOut: () => setIsPress(false),
    onPressIn: () => setIsPress(true),
  };

  var touchPropsSecondary = {
    activeOpacity: 1,
    underlayColor: Colors.white,
    onPressOut: () => setIsPress(false),
    onPressIn: () => setIsPress(true),
  };

  return (
    <View style={style}>
      {type === 'primary' && (
        <TouchableHighlight
          testID={testID}
          accessibilityLabel={
            accessibilityLabelText ? accessibilityLabelText : undefined
          }
          accessibilityHint={
            accessibilityHintText ? accessibilityHintText : undefined
          }
          importantForAccessibility={
            disabled === true ? 'no-hide-descendants' : 'auto'
          }
          accessibilityElementsHidden={disabled === true ? true : false}
          onPress={onPress}
          disabled={isDisabled}
          {...touchPropsPrimary}
          style={[
            styles.common,
            commonStyle,
            styles.primaryContainer,
            isDisabled ? disabledView : {},
            isPress
              ? styles.borderOnPress
              : isDisabled
              ? styles.borderDisabled
              : styles.border,
          ]}
          accessibilityRole={'button'}>
          <View style={styles.flexRow}>
            {icon !== '' && (
              <BoschIcon
                size={20}
                name={icon}
                color={Colors.white}
                style={styles.icon}
              />
            )}
            <CustomText
              allowFontScaling={true}
              style={{...textStyle}}
              text={text.toString()}
              color={Colors.white}
            />
          </View>
        </TouchableHighlight>
      )}
      {type === 'secondary' && (
        <TouchableHighlight
          testID={testID}
          accessibilityLabel={
            accessibilityLabelText ? accessibilityLabelText : undefined
          }
          accessibilityHint={
            accessibilityHintText ? accessibilityHintText : undefined
          }
          importantForAccessibility={
            disabled === true ? 'no-hide-descendants' : 'auto'
          }
          accessibilityElementsHidden={disabled === true ? true : false}
          onPress={onPress}
          disabled={isDisabled}
          {...touchPropsSecondary}
          style={[
            styles.common,
            isDisabled ? disabledView : {},
            commonStyle,
            isPress
              ? styles.borderOnPress
              : isDisabled
              ? styles.borderDisabled
              : styles.border,
          ]}
          accessibilityRole={'button'}>
          <View style={styles.flexRow}>
            {icon !== '' && (
              <BoschIcon
                size={20}
                name={icon}
                color={
                  isPress
                    ? Colors.blueOnPress
                    : isDisabled
                    ? Colors.grayDisabled
                    : Colors.darkBlue
                }
                style={styles.icon}
              />
            )}
            <CustomText
              allowFontScaling={true}
              style={{...textStyle}}
              text={text.toString()}
              color={
                isPress
                  ? Colors.blueOnPress
                  : isDisabled
                  ? Colors.grayDisabled
                  : Colors.darkBlue
              }
            />
          </View>
        </TouchableHighlight>
      )}
      {type === 'tertiary' && (
        <TouchableHighlight
          testID={testID}
          accessibilityLabel={
            accessibilityLabelText ? accessibilityLabelText : undefined
          }
          accessibilityHint={
            accessibilityHintText ? accessibilityHintText : undefined
          }
          importantForAccessibility={
            disabled === true ? 'no-hide-descendants' : 'auto'
          }
          accessibilityElementsHidden={disabled === true ? true : false}
          onPress={onPress}
          disabled={isDisabled}
          {...touchPropsSecondary}
          style={[styles.tertiary, isDisabled ? disabledView : {}]}
          accessibilityRole={'button'}>
          <View style={styles.flexRow}>
            {icon !== '' && (
              <BoschIcon
                size={20}
                name={icon}
                color={
                  isPress
                    ? Colors.blueOnPress
                    : isDisabled
                    ? Colors.grayDisabled
                    : Colors.darkBlue
                }
                style={styles.icon}
              />
            )}
            <CustomText
              allowFontScaling={true}
              style={{...textStyle}}
              text={text.toString()}
              color={
                isPress
                  ? Colors.blueOnPress
                  : isDisabled
                  ? Colors.grayDisabled
                  : Colors.darkBlue
              }
            />
          </View>
        </TouchableHighlight>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  common: {
    paddingVertical: 15,
    marginVertical: 5,
    alignItems: 'center',
  },
  border: {
    borderColor: Colors.darkBlue,
    borderWidth: 1,
  },
  borderOnPress: {
    borderColor: Colors.blueOnPress,
    borderWidth: 1,
  },
  borderDisabled: {
    borderColor: Colors.grayDisabled,
    borderWidth: 1,
  },
  icon: {
    paddingRight: 5,
    height: 20,
  },
  flexRow: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  primaryContainer: {
    backgroundColor: Colors.darkBlue,
  },
  tertiary: {
    paddingVertical: 10,
    alignItems: 'center',
  },
});
