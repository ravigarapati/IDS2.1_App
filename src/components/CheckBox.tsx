import React, { useState } from 'react';
import BoschIcon from './BoschIcon';
import {TouchableOpacity, StyleSheet, View} from 'react-native';
import {Colors} from '../styles';
import CustomText from './CustomText';
import {Icons} from '../utils/icons';

type checkBoxProps = {
  checked: boolean;
  onChange?: any;
  style?: {};
  textStyle?: {};
  text?: string;
  props?: any;
  accessibilityLabel?: string;
  lines?: number;
  disabled?: boolean;
};

export default function CheckBox({
  checked,
  onChange,
  style,
  textStyle,
  text,
  accessibilityLabel,
  lines,
  disabled = false,
  ...props
}: checkBoxProps) {

  const [showText, setShowText] = useState(false) 

  return (
    <View style={[styles.container, style]} {...props}>
      <TouchableOpacity
        onPress={() => {
          !disabled ? onChange(!checked) : null;
        }}
        style={text && styles.marginRight10}
        hitSlop={{top: 20, bottom: 20, left: 20, right: 20}}
        accessibilityLabel={accessibilityLabel}
        accessibilityState={{checked: checked}}>
        <View
          style={[
            checked ? styles.checkBoxSelected : styles.checkBoxUnselected,
          ]}>
          {checked && (
            <BoschIcon name={Icons.checkmark} size={24} color={Colors.white} style={{height: 24}} />
          )}
        </View>
      </TouchableOpacity>
      {text && 
      <View>
      {!lines &&<CustomText style={textStyle} text={text} align="left" />}
      {lines &&( 
      <View>  
        <CustomText style={textStyle} text={text} noOfLines={showText ? undefined : lines} align="left" />
        <TouchableOpacity onPress={() => setShowText(!showText)}>
          <CustomText style={textStyle} color={Colors.mediumBlue} text={showText ? 'Show less' : 'Show more' } align="left" />
        </TouchableOpacity>
      </View>
        )}
        </View>}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: Colors.white,
    marginVertical: 20,
  },
  checkBoxSelected: {
    backgroundColor: Colors.darkBlue,
    width: 24,
    height: 24,
  },
  checkBoxUnselected: {
    backgroundColor: Colors.mediumGray,
    width: 24,
    height: 24,
  },
  marginRight10: {
    marginRight: 10,
  },
});
