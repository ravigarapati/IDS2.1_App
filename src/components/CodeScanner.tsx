import React from 'react';
import {Dimensions, StyleSheet, View, Text} from 'react-native';
import QRCodeScanner from 'react-native-qrcode-scanner';
import BarcodeMask from 'react-native-barcode-mask';
import {Colors} from '../styles';
import Button from './Button';
import {Dictionary} from '../utils/dictionary';
// import {RNCamera} from 'react-native-camera';
type scannerProps = {
  onClose: any;
  data: any;
  onCancel?: any;
  textToDisplay?: string;
  height?: any;
  testID?:string;
};

export default function CodeScanner({
  onClose,
  data,
  onCancel,
  textToDisplay,
  height = undefined,
  testID,
}: scannerProps) {
  return (
    <View testID={testID} style={{flex: 1}}>
      {textToDisplay && (
        <Text
          style={{
            position: 'absolute',
            alignSelf: 'center',
            bottom: '25%',
            opacity: 0.7,
            zIndex: 100,
          }}>
          {textToDisplay}
        </Text>
      )}
      <QRCodeScanner
        reactivate={true}
        reactivateTimeout={2000}
        onRead={value => {
          data(value.data);
          onClose(false);
        }}
        checkAndroid6Permissions={true}
        cameraStyle={
          height ? {height: height} : {height: Dimensions.get('window').height}
        }
        // flashMode={RNCamera.Constants.FlashMode.torch}
        showMarker={true}
        markerStyle={styles.QRstyle}
      />
      <BarcodeMask
        edgeColor={Colors.white}
        showAnimatedLine
        backgroundColor={Colors.lightGray}
        edgeBorderWidth={3}
      />

      <Button
        type="secondary"
        onPress={() => (onCancel ? onCancel(false) : onClose(false))}
        style={styles.closeButton}
        text={Dictionary.button.cancel}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  closeButton: {
    backgroundColor: Colors.white,
    padding: 20,
  },
  QRstyle: {
    borderWidth: 0,
  },
});
