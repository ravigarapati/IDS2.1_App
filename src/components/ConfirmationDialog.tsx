import React from 'react';
import { Modal, StyleSheet, View } from 'react-native';
import Button from './Button';
import CustomText from './CustomText';
import { Colors } from '../styles';
import { BoschIcon } from '../components';
import { Icons } from '../utils/icons';

type ConfirmationDialogProps = {
  title?: string;
  text: string;
  textAlign?: any;
  textColor?: any;
  warningAlertText?: string;
  primaryButton?: string;
  secondaryButton?: string;
  primaryButtonOnPress?: () => void;
  secondaryButtonOnPress?: () => void;
  style?: any;
  visible: boolean;
};

export default function ConfirmationDialog({
  title,
  text,
  textAlign,
  textColor,
  warningAlertText,
  primaryButton,
  secondaryButton,
  primaryButtonOnPress,
  secondaryButtonOnPress,
  visible,
}: ConfirmationDialogProps) {
  return (
    <Modal animationType="none" transparent={true} visible={visible}>
      <View style={styles.modalContainer}>
        <View style={styles.contentContainer}>
          {title ? (
            <CustomText
              text={title}
              align="center"
              color={textColor ? textColor : Colors.black}
              style={styles.marginBottom}
            />
          ) : null}
          <CustomText text={text} align={textAlign ? textAlign : 'left'} color={textColor ? textColor : Colors.black} />
          {warningAlertText ? (
            <View style={styles.noteAlert}>
              <BoschIcon
                name={Icons.alertWarning}
                size={36}
                color="#725B00"
                style={styles.icon}
              />
              <CustomText
                text={warningAlertText}
                align="left"
                color="#725B00"
              />
            </View>
          ) : null}
          <View style={[styles.paddingVertical10]}>
            {primaryButton && (
              <Button
                type="primary"
                text={primaryButton}
                onPress={primaryButtonOnPress}
              />
            )}
            {secondaryButton && (
              <Button
                type="secondary"
                text={secondaryButton}
                onPress={secondaryButtonOnPress}
              />
            )}
          </View>
        </View>
      </View>
    </Modal>
  );
}
const styles = StyleSheet.create({
  container: {
    height: '100%',
  },
  contentContainer: {
    width: '90%',
    borderWidth: 0.5,
    borderColor: Colors.mediumGray,
    backgroundColor: Colors.white,
    padding: 20,
    shadowColor: Colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  paddingVertical10: {
    paddingVertical: 10,
  },
  marginBottom: {
    marginBottom: 20,
  },
  modalContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.blur,
  },
  noteAlert: {
    flexDirection: 'row',
    backgroundColor: '#FFDF95',
    color: '#725B00',
    padding: 15,
    marginBottom: 50,
    marginTop: 5
  },
  icon: {
    marginRight: 15,
    marginTop: 5,
    height: 36,
  }
});
