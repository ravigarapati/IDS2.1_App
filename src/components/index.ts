import ToggleButton from './ToggleButton';
import Button from './Button';
import CheckBox from './CheckBox';
import CustomText from './CustomText';
import CustomInputText from './CustomInputText';
import BoschIcon from './BoschIcon';
import ProgressIndicator from './ProgressIndicator';
import CustomHeader from './CustomHeader';
import CustomPicker from './CustomPicker';
import InfoTooltip from './InfoTooltip';
import Loader from './Loader';
import ConfirmationDialog from './ConfirmationDialog';
import CustomDialogBox from './CustomDialogBox';
import SectionHeading from './SectionHeading';
import CodeScanner from './CodeScanner';
import Banner from './Banner';
import Link from './Link';
import Accordion from './Accordion';
import SearchBar from './SearchBar';
import CreateProfileWalkthrough from './CreateProfileWalkthrough';
import ContractorHomeWalkthrough from './ContractorHomeWalkthrough';
import CustomAutoCompleteInput from './CustomAutoCompleteInput';
import DeviceAdded from './DeviceAdded';
import DevicesModal from './DevicesModal';
import Dropdown from './Dropdown';
import ModalComponent from './ModalComponent';
import Tile from './Tile';
import RadioButton from './Radiobutton';
import SwitchContent from './SwitchContent';
import CustomSettingsInput from './CustomSettingsInput';
import CustomPickerColumns from './CustomPickerColumns';
import CustomWheelPicker from './CustomWheelPicker';
import CustomWheelPickerColums from './CustomWheelPickerColums';
export {
  ToggleButton,
  Button,
  CheckBox,
  CustomText,
  CustomInputText,
  BoschIcon,
  ProgressIndicator,
  CustomHeader,
  CustomPicker,
  InfoTooltip,
  Loader,
  ConfirmationDialog,
  CustomDialogBox,
  SectionHeading,
  CodeScanner,
  Banner,
  Link,
  Accordion,
  SearchBar,
  CreateProfileWalkthrough,
  ContractorHomeWalkthrough,
  CustomAutoCompleteInput,
  DeviceAdded,
  DevicesModal,
  Dropdown,
  ModalComponent,
  Tile,
  RadioButton,
  SwitchContent,
  CustomSettingsInput,
  CustomPickerColumns,
  CustomWheelPicker,
  CustomWheelPickerColums,
};
