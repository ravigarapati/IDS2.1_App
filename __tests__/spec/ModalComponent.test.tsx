import 'react-native';
import React from 'react';
import {ModalComponent} from '../../src/components';
import renderer from 'react-test-renderer';

describe('Modal Component', () => {
  test('Component to be defined', () => {
    const tree = renderer
      .create(
        <ModalComponent
          modalVisible={true}
          closeModal={() => {}}
          children={null}
        />,
      )
      .toJSON();
    expect(tree).toBeDefined();
  });
  test('Component to be rendered', () => {
    const tree = renderer
      .create(
        <ModalComponent
          modalVisible={true}
          closeModal={() => {}}
          children={null}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
