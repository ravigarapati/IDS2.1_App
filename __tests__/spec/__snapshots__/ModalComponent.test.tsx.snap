// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component Component to be rendered 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={false}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "stretch",
            "backgroundColor": "white",
            "elevation": 5,
            "margin": 20,
            "paddingBottom": 35,
            "paddingHorizontal": 10,
            "paddingTop": 16,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      />
    </View>
  </View>
</Modal>
`;
